### ISSUE TITLE
ConnectivityManager leak in MainActivity 

### ISSUE THREAD

**Issue by tobrun** [mapbox/mapbox-gl-native#3525]
![device-2016-01-12-153541](https://cloud.githubusercontent.com/assets/2151639/12267014/630c0982-b946-11e5-8d5c-53878c91026f.png)

**Comment by tobrun** [comment]
I believe this is rather an Android SDK issue than a Mapbox one..

**Comment by tobrun** [comment]
I have created an issue on [issue tracker of LeakCanary](https://github.com/square/leakcanary/issues/393) as in the [AOSP issue tracker](https://code.google.com/p/android/issues/detail?id=198852&sort=-opened&colspec=ID%20Status%20Priority%20Owner%20Summary%20Stars%20Reporter%20Opened)

**Comment by tobrun** [comment]
There is currently nothing actionable to do for us. closing.

**Comment by tobrun** [comment]
I have identified that this line is the problem:
```java
ConnectivityManager connectivityManager = (ConnectivityManager) context.getSystemService(Context.CONNECTIVITY_SERVICE);
```

This leak can be prevented if we replace above with:
```java
ConnectivityManager connectivityManager = (ConnectivityManager) context.getApplicationContext()
                .getSystemService(Context.CONNECTIVITY_SERVICE);
```

**Comment by tobrun** [comment]
This will land in the release branch of `3.1.0` in #3537.

**Comment by tobrun** [comment]
This landed with 5ff95d4 in branch mentioned above.

**Comment by saiidabs** [comment]
@tobrun  thanks man , i was stuck on this problem , life saver
